cranelift/isle/docs/language-reference.md

```diff
@@ -1430,10 +1430,30 @@ for more information regarding the trie construction algorithm.
 
 ## Reference: ISLE Language Grammar
 
-Baseline: allow arbitrary whitespace, and Lisp-style comments (`;` to
-newline). The grammar accepted by the parser is as follows:
+Baseline: allow arbitrary whitespace, and wasm-style comments (`;` to
+newline, or nested block-comments with `(;` and `;)`).
+
+The grammar accepted by the parser is as follows:
 
 ```bnf
+<skip> ::= <whitespace> | <comment>
+
+<whitespace> ::= " "
+               | "\t"
+               | "\n"
+               | "\r"
+
+<comment> ::= <line-comment> | <block-comment>
+
+<line-comment> ::= ";" <line-char>* (<newline> | eof)
+<newline> ::= "\n" | "\r"
+<line-char> ::= <any character other than "\n" or "\r">
+
+<block-comment> ::= "(;" <block-char>* ";)"
+<block-char> ::= <any character other than ";" or "(">
```

Comment:
It looks like the parser below supports nested block comments; can we represent this in the BNF?

(Maybe something like `<block-char> += <block-comment>`, i.e., a nested comment is one `<block-char>` in the parse tree)

Reply:
The nesting is represented by the `| <block-comment>` alternative. I copied pretty much directly from the wasm spec: https://webassembly.github.io/spec/core/text/lexical.html#comments

Reply:
Ah, I missed the last option -- very sorry. Thanks, this looks right!